src/components/SingleArticle.jsx

```diff
@@ -1,32 +1,47 @@
 import { useState, useEffect } from "react";
 import { useParams } from "react-router-dom";
-import { fetchArticleById } from "../../utils/api";
+import { getArticleById, getCommentsByArticleId } from "../../utils/api";
+import CommentCard from "./CommentCard";
 
 function SingleArticle() {
   const { article_id } = useParams();
   const [article, setArticle] = useState({});
+  const [comments, setComments] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
     setIsLoading(true);
-    fetchArticleById(article_id).then((response) => {
+    getArticleById(article_id).then((response) => {
       setArticle(response);
       setIsLoading(false);
     });
+    getCommentsByArticleId(article_id).then((response) => {
+      setComments(response);
+    });
   }, [article_id]);
 
   if (isLoading) return <p>Loading...</p>;
```

Comment:
We should have a separate loading screen for the articles and the comments so the user does not wait for all the comments to render to read the article 🙂

Reply:
fixed in #5